src/main/java/org/gridsuite/modification/server/modifications/GenerationDispatch.java

```diff
@@ -0,0 +1,223 @@
+        double totalLoad = computeTotalActiveLoad(component);
+        return totalLoad * (1. + lossCoefficient / 100.);
+    }
+
+    private double computeTotalAmountFixedSupply(int numCC) {
+        double totalAmountFixedSupply = 0.;
+        totalAmountFixedSupply += fixedSupplyGenerators.get(numCC).stream().filter(generator -> generator.getTerminal().isConnected())
+            .mapToDouble(Generator::getTargetP).sum();
+        return totalAmountFixedSupply;
+    }
+
+    private void computeAdjustableGenerators(Component component, Reporter reporter) {
+        List<Generator> generators = new ArrayList<>();
+
+        // get all connected generators in the component
+        for (Bus bus : component.getBuses()) {
+            generators.addAll(bus.getGeneratorStream().filter(generator -> generator.getTerminal().isConnected())
+                .collect(Collectors.toList()));
+        }
+        // remove non adjustable generators (empty list in this first version)
+        generators.removeAll(fixedSupplyGenerators.get(component.getNum()));
+
+        // remove generators without marginal cost
+        adjustableGenerators.put(component.getNum(), generators.stream().filter(generator -> {
+            GeneratorStartup startupExtension = generator.getExtension(GeneratorStartup.class);
+            boolean marginalCostAvailable = startupExtension != null && !Double.isNaN(startupExtension.getMarginalCost());
+            if (!marginalCostAvailable) {
+                report(reporter, "MissingMarginalCostForGenerator", "The generator ${generator} does not have a marginal cost",
+                    Map.of("generator", generator.getId()), TypedValue.WARN_SEVERITY);
+            }
+            return marginalCostAvailable;
+        }).collect(Collectors.toList()));
+
+        // sort generators by marginal cost, and then by alphabetic order of id
+        adjustableGenerators.get(component.getNum()).sort(Comparator.comparing(generator -> ((Generator) generator).getExtension(GeneratorStartup.class).getMarginalCost())
+            .thenComparing(generator -> ((Generator) generator).getId()));
+
+        if (adjustableGenerators.get(component.getNum()).isEmpty()) {
+            report(reporter, "NoAvailableAdjustableGenerator", "There is no adjustable generator",
+                Map.of(), TypedValue.WARN_SEVERITY);
+        }
+    }
+
+    private static class GeneratorTargetPListener extends DefaultNetworkListener {
+        private final Reporter reporter;
+
+        GeneratorTargetPListener(Reporter reporter) {
+            this.reporter = reporter;
+        }
+
+        @Override
+        public void onUpdate(Identifiable identifiable, String attribute, String variantId, Object oldValue, Object newValue) {
+            if (identifiable.getType() == IdentifiableType.GENERATOR &&
+                attribute.equals("targetP") &&
+                (double) oldValue != (double) newValue) {
```

Comment:
I think, by definition, old is different from new in case of update

Reply:
it depends on the implementation : in network-store, this is the case, but not in powsybl-core implementation
So I prefer to test it here ....